src/main/java/org/kaazing/netx/ws/internal/io/OutgoingSentinelExtension.java

```diff
@@ -262,7 +262,7 @@ else if (reasonLength > 123) {
 
             int reasonOffset = closePayload.reasonOffset();
 
-            connection.getRandom().nextBytes(mask);
+            createMask(connection, mask);
```

Comment:
Also, wouldn't we want to fill the mask directly into the output buffer instead of keeping it separate and copying it later?